### PR TITLE
chore(lint): add custom rules for selectors

### PR DESCRIPTION
**Pull request by Nodonisko**
Few custom rules that will prevent dangerous mistakes while using getState and selectors in new packages.

### CODE REVIEW

File: suite-common/redux-utils/src/createThunk.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
-import { AsyncThunkOptions, AsyncThunkPayloadCreator, createAsyncThunk } from '@reduxjs/toolkit';
+import {
+    AsyncThunkOptions,
+    AsyncThunkPayloadCreator,
+    createAsyncThunk as createAsyncThunkReduxToolkit,
```
Comment (juriczech) [inline suite-common/redux-utils/src/createThunk.ts:4]: Why rename it if there is no overshadow of variable names?

**Review by juriczech — APPROVED** [review]

File: .eslintrc.js [hunk 2]
```diff
@@ -182,6 +185,32 @@ module.exports = {
...
+            {
+                message:
+                    "Please don't use createAsyncThunk. Use createThunk from @suite-common/redux-utils instead.",
+                selector: "CallExpression[callee.name='createAsyncThunk']",
+            },
+            {
+                message:
+                    'Please don\'t use getState directly. Always use strongly typed selector, because geState is typed as "any" and it\'s dangerous to use it directly.',
+                selector:
+                    'MemberExpression[property.type="Identifier"]:matches([object.callee.name="getState"])',
```
Comment (bouchja1) [inline .eslintrc.js:199]: There are many cases in suite actions that have to be modified to selectors according to this rule, am I right? E.g. in router actions, metadata actions etc.

File: suite-common/connect-init/src/slice.ts [hunk 3]
```diff
@@ -37,13 +33,7 @@ export const prepareConnectInitThunk = ({
...
     initSettings: { manifest: Manifest } & Partial<ConnectSettings>;
 }) =>
-    createAsyncThunk<
-        void,
-        void,
-        {
-            state: any;
-        }
-    >(CONNECT_INIT_ACTION_TYPE, async (_, { dispatch, getState }) => {
+    createThunk(CONNECT_INIT_ACTION_TYPE, async (_, { dispatch, getState }) => {
```
Comment (bouchja1) [inline suite-common/connect-init/src/slice.ts:36]: Btw. does it make sense to have `prepareConnectInitThunk` still here when we have extra dependencies concept with redux-utils implemented now? `actions` and `selectors` can be moved to extra under these keys sections and initSettings moved to `utils` (platform-specific things) maybe?

Comment (Nodonisko) [reply suite-common/connect-init/src/slice.ts:36]: It is not, but want to keep this simple. There will be follow up

File: .eslintrc.js (hunk 2 above)
Comment (Nodonisko) [reply .eslintrc.js:199]: Everything needs to be selector, but we will migrate it during migration to common

File: suite-common/redux-utils/src/createThunk.ts (hunk 1 above)
Comment (Nodonisko) [reply suite-common/redux-utils/src/createThunk.ts:4]: Because it will throw error otherwise, because of new eslint rule 😅

**Pull request merged**